runtime/serde/serde-json/common/src/aws/smithy/kotlin/runtime/serde/json/JsonSerializer.kt

```diff
@@ -197,6 +197,11 @@ public class JsonSerializer : Serializer, ListSerializer, MapSerializer, StructS
         if (value != null) serializeInstant(value, format) else jsonWriter.writeNull()
     }
 
+    override fun entry(key: String, value: Document?) {
+        jsonWriter.writeName(key)
+        if (value != null) serializeDocument(value) else jsonWriter.writeNull()
```

Comment:
The null check actually isn't necessary here, `serializeDocument()` handles it since `null` is a valid value of a document. It's fine as is though.

Reply:
Oh good point, I'll clean it up to reduce code duplication.